crypto/kernel/err.c

```diff
@@ -67,12 +68,33 @@ srtp_err_status_t srtp_err_reporting_init ()
     return srtp_err_status_ok;
 }
 
+static srtp_err_report_handler_func_t * srtp_err_report_handler = NULL;
+
+srtp_err_status_t srtp_install_err_report_handler(srtp_err_report_handler_func_t func)
+{
+    srtp_err_report_handler = func;
+    return srtp_err_status_ok;
+}
+
 void srtp_err_report (srtp_err_reporting_level_t level, const char *format, ...)
 {
     va_list args;
-    va_start(args, format);
     if (srtp_err_file != NULL) {
+        va_start(args, format);
         vfprintf(srtp_err_file, format, args);
+        va_end(args);
+    }
+    if (srtp_err_report_handler != NULL) {
+        va_start(args, format);
+        char msg[512];
```

Comment:
Should this msg buffer be considered a storage of secrets which should be cleared using `octet_string_set_to_zero()` after being used? Honestly not sure. :-)